FWCore/Framework/interface/GlobalSchedule.h

```diff
@@ -213,7 +214,11 @@ namespace edm {
             }
             iHolder.doneWaiting(excpt);
           });
-      WorkerManager& workerManager = workerManagers_[principal.index()];
+      unsigned int managerIndex = principal.index();
+      if constexpr (T::branchType_ == InRun) {
+        managerIndex += numberOfConcurrentLumis_;
+      }
+      WorkerManager& workerManager = workerManagers_[managerIndex];
```

Comment:
Yes. This seemed easier because there are a few places where we iterate over all the worker managers. If we put the WorkerManagers in two separate containers then those would need to be two separate loops. I don't feel strongly about this though. I almost did it the other way. If you ask, I will change it.

Reply:
It wasn't meant as a request for change, it was a design note to help me understand the change and to essentially ask if I understood the change.